packages/webview_flutter/webview_flutter_android/CHANGELOG.md

```diff
@@ -1,3 +1,8 @@
+## 4.1.1
+
+* Explicitly set WebView Settings `setAllowFileAccess` to `false` by default for Android SDK versions less than 29 to enhance security.
```

Comment:
I don't believe this would be considered as a breaking change. The reason I say this is because when the function `loadFile` is called, it sets `allowFileAccess` as `true`. This is done intentionally assuming that the `allowFileAccess` is set to `false` by default. However as pointed out in another comment, that the documentation is not accurate.

> I don't see a compelling reason for webview_flutter to make a breaking change to be more strict that the platform has chosen to be, especially since it sounds from the documentation like the benefit would only be to developers who are ignoring Android best practices of targeting the latest SDK.

> In fact, given [Play Store policy](https://support.google.com/googleplay/android-developer/answer/11917020?hl=en), it seems like this change would only benefit apps that aren't being published, and haven't been published in a couple of years at least, which definitely would not warrant a breaking change.

It would also benefit developers that have applications minSdkVersion targeting [Build.VERSION_CODES.Q](https://developer.android.com/reference/android/os/Build.VERSION_CODES#Q). This is common for applications that were created a while back.

In any case, I would still assume it would be in the plugin's best interest to follow best practices as noted by the document:
>To prevent possible security issues targeting [Build.VERSION_CODES.Q](https://developer.android.com/reference/android/os/Build.VERSION_CODES#Q) and earlier, you should explicitly set this value to false.

Reply:
> I don't believe this would be considered as a breaking change. The reason I say this is because when the function `loadFile` is called, it sets `allowFileAccess` as `true`.

It is a breaking change, because `loadFile` is not the only entry point to load content. If it were, this part of the PR wouldn't do anything.

The sole purpose of changing the default is to cause some loads that would currently succeed to instead fail. That's a breaking change. Ideally nobody is relying on the behavior the change would break, but we have no way of knowing that.

> In any case, I would still assume it would be in the plugin's best interest to follow best practices as noted by the document:
> 
> > To prevent possible security issues targeting [Build.VERSION_CODES.Q](https://developer.android.com/reference/android/os/Build.VERSION_CODES#Q) and earlier, you should explicitly set this value to false.

That documentation is client-facing. `webview_flutter` is not conceptually a client of `WebView`, it is a wrapper; the client is the client of `webview_flutter`. Exposing the new API allows clients to follow those best practices if that's the right behavior for their use case.

Reply:
> It is a breaking change, because `loadFile` is not the only entry point to load content. If it were, this part of the PR wouldn't do anything.
>
>The sole purpose of changing the default is to cause some loads that would currently succeed to instead fail. That's a breaking change. Ideally nobody is relying on the behavior the change would break, but we have no way of knowing that.

I may not be fully understanding what you are stating here. Is the value for `allowFileAccess` not by default `false` anyways? If so, then why would it matter if it was explicitly set.

>That documentation is client-facing. webview_flutter is not conceptually a client of WebView, it is a wrapper; the client is the client of webview_flutter. Exposing the new API allows clients to follow those best practices if that's the right behavior for their use case.

Oh I understand, in this case you are suggesting that we do not explicitly set the default value, just expose the API to set the `allowFileAccess`, this way clients can follow the best practices as they see fit depending on their app?

Reply:
> I may not be fully understanding what you are stating here. Is the value for `allowFileAccess` not by default `false` anyways?

You explicitly said [here](https://github.com/flutter/packages/pull/8228#issuecomment-2523323456) that it is not, on devices running older versions of Android.

And even ignoring that, one of the following must be true:
1) The default is always `false`, in which case [this line](https://github.com/flutter/packages/pull/8228/files#diff-b8553801076e23e2d7f9ad3c518e4a21b0aa09df0741b992e4c2a5e86629e5e2R92) is a no-op and there is no reason to add it, or
2) The default is not always `false`, in which case that same line is a breaking change.

Either way, I do not think we should add it.

> Oh I understand, in this case you are suggesting that we do not explicitly set the default value, just expose the API to set the `allowFileAccess`, this way clients can follow the best practices as they see fit depending on their app needs?

Yes. Only the client knows whether or not they can safely disable file access.

Reply:
Good point, I have made the changes to only expose the `setAllowFileAccess`. 

Thank you!